### PR TITLE
integrate with prepared geom

### PR DESCRIPTION
**Pull request by michaelkirk**
I'm unfamiliar with R, and wasn't able to figure out how to run the R benchmarks.

There's an unfortunate clone, but I'd be curious to know if is this is an improvement for you.

### CODE REVIEW

File: src/rust/src/spatial_index.rs [hunk 1]
```diff
@@ -49,13 +50,15 @@ pub fn create_cached_rtree(geoms: List) -> RTree<GeomWithData<CachedEnvelope<Geo
                 None
             } else {
                 let geo = Geom::try_from(xi).unwrap();
+                let prepared_geom: PreparedGeometry<'static> = PreparedGeometry::from(geo.geom.clone());
```
Comment (michaelkirk) [inline src/rust/src/spatial_index.rs:53]: heres' the unfortunate clone. There should be a way to avoid this, but I haven't worked that out yet.

**Comment by michaelkirk** [comment]
Also note that I pushed up a new commit to https://github.com/georust/geo/tree/mkirk/prepared-geom to allow you to build `PreparedGeometry::from(geometry)`. Previously, it only supported `PreparedGeometry::from(polygon)`.

File: src/rust/src/spatial_index.rs (hunk 1 above)
Comment (JosiahParry) [reply src/rust/src/spatial_index.rs:53]: It looks like the AABB is still being calculated from the cached envelope of the `Geom` struct. Is there a way to get it from the prepared geometry?

Comment (JosiahParry) [reply src/rust/src/spatial_index.rs:53]: Or is it possible to completely ignore the R* tree because of the prepared geoms? 🤔

**Comment by JosiahParry** [comment]
I can't find a PR for the PreparedGeometry so I'll leave some here. I think the two things that confuse me are

1. how to interact with methods that are applied to the `Geometry` enum 
2. the lifetimes. Should i be slapping `'static` everywhere? Would it be possible to have `PreparedGeometry<T>` for all Geometry types?

**Pull request closed without merging**